examples/python/whisper.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License
+
+import argparse
+import glob
+import os
+import readline
+
+import onnxruntime_genai as og
+
+
+def _complete(text, state):
+    return (glob.glob(text + "*") + [None])[state]
+
+
+class Format:
+    end = "\033[0m"
+    underline = "\033[4m"
+
+
+def run(args: argparse.Namespace):
+    print("Loading model...")
+    model = og.Model(args.model_path)
+    processor = model.create_multimodal_processor()
```

Comment:
Should we consider creating a processor as `og.Processor(model)` in the future similar to how `og.Tokenizer(model)` exists?

Reply:
Whisper is a typical encoder-decoder model. It is different with the Multi-Modal model. the process i am thinking:

prompt = "<prev>prev_text<SOT><en><transcrib>"
prompt_tokens = Processor.encode(prompt)
encoder_input = Processor.ProcessAudio(audio);

generator.append(prompt_tokens)
generaotr.set_encoder_input(encoder_input)

Reply:
Many models have very blurred lines between the encoder and the decoder (such as the phi3 vision model).
I don't think the user wants to know whether a model is encoder-decoder model or if it is only a decoder model (or if it something else).

Unless we want to expose ways for the user to manipulate the output of the encoder before it is fed back into the decoder, we can potentially hide this complexity from the user.

For now, this pull-request separates out the encoder and decoder inputs and the user must explicitly set both the encoder as well as the decoder inputs by themselves.

We should discuss and review this design choice in a meeting/scrum when possible.